AK/IpAddressCidr.h

```diff
@@ -191,8 +198,96 @@ struct Formatter<IPv4AddressCidr> : Formatter<StringView> {
 };
 #endif
 
+class IPv6AddressCidr : public Details::IPAddressCidr<IPv6AddressCidr> {
+public:
+    constexpr IPv6AddressCidr(IPv6Address address, u8 length)
+        : IPAddressCidr(address, length)
+    {
+    }
+
+    constexpr IPv6Address first_address_of_subnet() const
+    {
+        u8 address[16] = { 0 };
+        u8 free_bits = MAX_LENGTH - length();
+
+        if (free_bits != 128) {
+            NetworkOrdered<u128> mask = NumericLimits<u128>::max() << free_bits;
+            auto address_mask = bit_cast<Array<u8, 16>>(mask);
+
+            auto const* original_address = ip_address().to_in6_addr_t();
+            for (int i = 0; i < 16; i++) {
+                address[i] = original_address[i] & address_mask[i];
+            }
+        }
+
+        return address;
+    }
+
+    constexpr IPv6Address last_address_of_subnet() const
+    {
+        u8 address[16] = { 0 };
+        u8 free_bits = MAX_LENGTH - length();
+
+        NetworkOrdered<u128> inverse_mask = NumericLimits<u128>::max() >> (128 - free_bits);
+
+        if (free_bits != 128) {
+            NetworkOrdered<u128> mask = NumericLimits<u128>::max() << free_bits;
+            auto address_mask = bit_cast<Array<u8, 16>>(mask);
+
+            auto const* original_address = ip_address().to_in6_addr_t();
+            for (int i = 0; i < 16; i++) {
+                address[i] = original_address[i] & address_mask[i];
+            }
+        }
+
+        auto inverse_address_mask = bit_cast<Array<u8, 16>>(inverse_mask);
+
+        for (int i = 0; i < 16; i++) {
+            address[i] = address[i] | inverse_address_mask[i];
+        }
+
+        return address;
+    }
+
+    bool contains(IPv6Address other) const
+    {
+        IPv6AddressCidr other_cidr = IPv6AddressCidr::create(other, length()).value();
+        return first_address_of_subnet() == other_cidr.first_address_of_subnet();
+    }
+
+    static u8 const MAX_LENGTH = 128;
+};
+
+template<>
+struct Traits<IPv6AddressCidr> : public DefaultTraits<IPv6AddressCidr> {
+    static unsigned hash(IPv6AddressCidr const& address)
+    {
+        IPv6Address ip_address = address.ip_address();
+        return sip_hash_bytes<4, 8>({ &ip_address, address.length() });
+    }
+};
+
+#ifdef KERNEL
+template<>
+struct Formatter<IPv6AddressCidr> : Formatter<StringView> {
+    ErrorOr<void> format(FormatBuilder& builder, IPv6AddressCidr value)
+    {
+        return Formatter<StringView>::format(builder, TRY(value.to_string())->view());
+    }
+};
+#else
+template<>
+struct Formatter<IPv6AddressCidr> : Formatter<StringView> {
+    ErrorOr<void> format(FormatBuilder& builder, IPv6AddressCidr value)
+    {
+        return Formatter<StringView>::format(builder, TRY(value.to_string()));
+    }
+};
```

Comment:
See the OG commit

Reply:
What I meant with this, is that this could mirror the actual implementation of the CIDR to_string method,
maybe even replace it.

With some template shenanigans (Ie `DerivedFrom<IpAddressCIDR>`) all formatters could use the same code